src/embed.js

```diff
@@ -25,13 +25,19 @@ var PREPROCESSOR = {
   'vega-lite': function(vljson) { return vl.compile(vljson).spec; }
 };
 
-function load(url, arg, el, callback) {
+function load(url, arg, prop, el, callback) {
   vg.util.load({url: url}, function(err, data) {
+    var opt = {};
     if (err || !data) {
       console.error(err || ('No data found at ' + url));
     } else {
       // marshal embedding spec and restart
-      var opt = !arg ? JSON.parse(data) : vg.util.extend({source: data}, arg);
+      if (!arg) { // Loading embed spec from URL
+        opt = JSON.parse(data);
+      } else {  // Loading vg/vl spec or config from URL
+        opt[prop] = JSON.parse(data);
+        opt = vg.util.extend({}, arg, opt);
```

Comment:
Aha, didn't think of that use-case with `source`. +1 to continuing to support it. And agreed on the ordering of `opt`, I must've had the jetlag brain when I wrote that! Force pushed an update w/both. Thanks!